content/en/blog/2025/opentelemetry-for-legacy-app/index.md

```diff
@@ -0,0 +1,148 @@
+
+The end goal is clear: we need to rewrite or migrate this application to a modern, scalable, and supportable platform. But a project like that takes months, or even years. What do we do in the meantime? We can’t operate in the dark.
+
+While OpenTelemetry is often celebrated for its role in modern, cloud-native architectures, its value doesn't stop there. In fact, it provides a powerful, often overlooked, solution for the very systems that aren't cloud-native. It acts as a bridge to modernization. By giving us some visibility into the legacy application today, we can de-risk its operation, plan for its replacement, and build the data-driven business case for its future.
+Let's walk through how to do this with a simulated legacy app, all without changing a single line of its code.
+
+## OpenTelemetry: Modern Observability for Legacy Code
+
+To simulate a common legacy pattern, we've built a simple application. A core executable written in C (simulating a system-level process) starts a Java Virtual Machine (JVM) to handle a specific task, like processing a transaction or a record.
+
+The flow is: 
+
+```bash
+C Application (legacy_app) -> starts JVM -> calls a Java method to process a task
+```
+
+When we run ./legacy_app, it works. But we can't answer critical questions: Is it keeping up with business demand? Is it about to run out of memory and crash during end-of-day processing? Is a slowdown in this app the root cause of last week's customer complaints?
+
+Let's find out.
+
+
+### Stage 1: Basic Health Monitoring (Is the System Stable?)
+
+Our first goal is to get vital signs monitor on our application. We need to see its CPU and memory usage to ensure it’s not about to fail. We can achieve this with the OpenTelemetry Java Agent by simply attaching it to the application using the `-javaagent` flag in the JVM startup arguments. In our example, we can do this using the `_JAVA_OPTIONS` environment variable.
+
+**Step 1: Set up the Environment**
+
+In our terminal, we'll configure the agent without touching the application itself.
+
+```bash
+# --- Part 1: Basic System Health Configuration ---
+
+# 1. Give our service a descriptive name
+export OTEL_SERVICE_NAME=legacy-part-processor
+
+# 2. Enable ONLY the metrics exporter
+export OTEL_METRICS_EXPORTER=otlp
+
+# 3. Point the agent to the collector's gRPC endpoint
+export OTEL_EXPORTER_OTLP_ENDPOINT=http://127.0.0.1:4317
+
+# 4. Specify OTLP protocol (grpc is often default, but explicit is clear)
+export OTEL_EXPORTER_OTLP_PROTOCOL=grpc
```

Comment:
I'm not necessarily saying you need to change this, but while the java SDK itself defaults to grpc/4317, the javaagent actually defaults to using http/protobuf on 4318 (see the note [here](https://opentelemetry.io/docs/zero-code/java/agent/configuration/#sdk-configuration)). At a minimum it _might_ make sense to remove the comment saying that grpc is the default, since it can be a little confusing